src/bindings/python/CMakeLists.txt

```diff
@@ -1,28 +1,41 @@
+include(ValhallaVersion)
+
 if (PREFER_EXTERNAL_DEPS)
-  find_package(pybind11 QUIET)
-  if (NOT pybind11_FOUND)
-    message(WARNING "No pybind11 found in system libraries, using vendored pybind11...")
+  find_package(nanobind QUIET CONFIG)
+  if (NOT TARGET nanobind::nanobind)
+    message(WARNING "No nanobind found in system libraries, using vendored nanobind...")
   endif()
 endif()
 
-if (NOT TARGET pybind11::pybind11_headers)
-  add_subdirectory(${VALHALLA_SOURCE_DIR}/third_party/pybind11 ${CMAKE_BINARY_DIR}/third_party/pybind11)
+if (NOT TARGET nanobind::nanobind)
+  add_subdirectory(${VALHALLA_SOURCE_DIR}/third_party/nanobind ${CMAKE_BINARY_DIR}/third_party/nanobind EXCLUDE_FROM_ALL)
 endif()
 
 message(STATUS "Installing python modules to ${Python_SITEARCH}")
 
-function(set_output_python rel_path)
-  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/${rel_path} ${CMAKE_CURRENT_BINARY_DIR}/${rel_path} COPYONLY)
-endfunction()
-
 # add and install module(s)
-pybind11_add_module(_valhalla ${CMAKE_CURRENT_SOURCE_DIR}/src/_valhalla.cc)
+nanobind_add_module(_valhalla 
+  NB_STATIC
+  STABLE_ABI
+  LTO
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/_valhalla.cc)
 target_link_libraries(_valhalla PRIVATE valhalla $<$<BOOL:${ENABLE_COVERAGE}>:gcov>)
-pybind11_add_module(graph_utils 
+nanobind_add_module(graph_utils
+  NB_STATIC
+  STABLE_ABI
+  LTO
   ${CMAKE_CURRENT_SOURCE_DIR}/src/graph_utils.cc
   ${CMAKE_CURRENT_SOURCE_DIR}/src/graph_id.cc)
 target_link_libraries(graph_utils PRIVATE valhalla $<$<BOOL:${ENABLE_COVERAGE}>:gcov>)
 
+if(APPLE)
+  set_target_properties(_valhalla PROPERTIES INSTALL_RPATH "@loader_path")
+  set_target_properties(graph_utils PROPERTIES INSTALL_RPATH "@loader_path")
```

Comment:
not entirely sure if this is needed, that's from docs IIRC. but auditwheel & friends rewrite the RPATH anyways